src/App.jsx

```diff
@@ -1,14 +1,95 @@
 import './App.css';
+// import ChatEntry from './components/ChatEntry';
+import ChatLog from './components/ChatLog';
+import data from './data/messages.json';
+import { useState } from 'react';
 
 const App = () => {
+  const [chatData, setChatData] = useState(data);
+  const [fontColor, setFontColor] = useState(null);
+
+  const senderList = [];
+  const sendersList = (chatData) => {
+    for (const chat of chatData) {
+      if (!senderList.includes(chat.sender)) {
+        senderList.push(chat.sender);
+      }
+    }
+    let title = '';
+    for (const sender of senderList) {
+      title += `${sender} and `;
+    }
+    return title.slice(0, -5);
+  };
+
+  const sendersTitle = sendersList(chatData);
+  const local = senderList[0];
+
+  const handleLike = (id) => {
+    setChatData((chatData) =>
+      chatData.map((chat) => {
+        if (chat.id === id) {
+          return { ...chat, liked: !chat.liked };
+        } else {
+          return chat;
+        }
+      })
+    );
+  };
```

Comment:
Nice work on this function! By passing this down to your individual chat messages you are now able to have a single source of truth!